packages/peregrine/lib/talons/CheckoutPage/useShippingMethod.js

```diff
@@ -81,28 +81,37 @@ export const useShippingMethod = props => {
         }
     }, [primaryShippingAddress]);
 
-    // Grab the selected shipping method from the primary shipping address.
+    // Grab the selected shipping method. In order:
+    // 1. From the results of our specific query to fetch it
+    // 2. The lowest cost shipping method
     const selectedShippingMethod = useMemo(() => {
+        let selectedMethod;
+
+        // From the results of our specific query to fetch it.
         try {
-            let selectedMethod =
-                primaryShippingAddress.selected_shipping_method;
-
-            if (!selectedMethod) {
-                // If there are shipping methods to choose from,
-                // pick the lowest cost one from there instead.
-                if (shippingMethods.length) {
-                    // We know that the shipping methods are sorted by price,
-                    // so the first one is the lowest cost one.
-                    selectedMethod = shippingMethods[0];
-                }
-            }
+            selectedMethod =
+                chosenShippingMethodData.cart.shipping_addresses[0]
+                    .selected_shipping_method;
+        } catch (err) {
+            // We don't have data from our specific query to fetch the selected shipping method.
+            // Intentionally swallow this error.
+        }
 
-            const { carrier_code, method_code } = selectedMethod;
-            return `${carrier_code}|${method_code}`;
-        } catch {
-            return null;
+        if (!selectedMethod) {
+            // From the lowest cost shipping method.
+            if (shippingMethods.length) {
+                // We sorted the shipping methods by price,
+                // so the first one is the lowest cost one.
+                selectedMethod = shippingMethods[0];
+                console.log(
+                    'selected method from list of methods',
+                    selectedMethod
+                ); // has all display data needed
```

Comment:
```suggestion
```

Reply:
Oops, thanks!

Fixed in https://github.com/magento/pwa-studio/commit/7318d2bd0602c7c5bbbacad19747c57247c7b0fa.